ts/src/index.ts

```diff
@@ -90,23 +101,37 @@ export async function initConfig(config: Config): Promise<ProfilerConfig> {
                     // ignore errors, which will occur when not on GCE.
                 }) ||
         [undefined, undefined];
-    if (!mergedConfig.zone && zone) {
-      mergedConfig.zone = zone.substring(zone.lastIndexOf('/') + 1);
+    if (!config.zone && zone) {
+      config.zone = zone.substring(zone.lastIndexOf('/') + 1);
     }
-    if (!mergedConfig.instance && instance) {
-      mergedConfig.instance = instance;
+    if (!config.instance && instance) {
+      config.instance = instance;
     }
   }
-
-  if (!hasService(mergedConfig)) {
-    throw new Error('Service must be specified in the configuration.');
-  }
-
-  return mergedConfig;
+  return config;
 }
 
 let profiler: Profiler|undefined = undefined;
 
+/**
+ * Initializes the config, and starts heap profiler if the heap profiler is
+ * needed.
+ */
+export async function startHelper(config: Config): Promise<ProfilerConfig> {
```

Comment:
Would be good to name it more specifically than `startHelper`. Maybe `prepareStart` or `preStart` or even `initConfigAndMaybeStartHeapProfiler`?

Reply:
I went with `initConfigAndMaybeStartHeapProfiler`.

Reply:
It doesn't look like the function is renamed? I'm still seeing startHelper.